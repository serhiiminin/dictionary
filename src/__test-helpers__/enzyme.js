import React from 'react';
import { shallow, mount } from 'enzyme';
import { ThemeProvider } from 'styled-components';

const shallowWithTheme = (tree, theme) => {
  const context = shallow(<ThemeProvider theme={theme} />)
    .instance()
    .getChildContext()
  return shallow(tree, { context })
}

const mountWithTheme = (tree, theme) => {
  const context = shallow(<ThemeProvider theme={theme} />)
    .instance()
    .getChildContext()

  return mount(tree, {
    context,
    childContextTypes: ThemeProvider.childContextTypes,
  })
}

export { shallowWithTheme, mountWithTheme }
