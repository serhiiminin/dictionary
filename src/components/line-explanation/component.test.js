import React from "react";
import { shallow, mount } from "enzyme";
import muiTheme from "../../root/mui-theme";
import LineExplanation, { LineExplanationWrapper } from ".";

describe("Line explanation wrapper", () => {
  test("render", () => {
    const wrapper = mount(
      <LineExplanationWrapper label="label" theme={muiTheme}>text</LineExplanationWrapper>
    );
    expect(wrapper).toMatchSnapshot();
  });
});

describe("Line explanation", () => {
  test("render", () => {
    const wrapper = shallow(
      <LineExplanation label="label" theme={muiTheme}>text</LineExplanation>
    );
    expect(wrapper).toMatchSnapshot();
  });
});
