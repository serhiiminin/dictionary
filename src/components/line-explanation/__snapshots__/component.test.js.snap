// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Line explanation render 1`] = `
<div
  className={undefined}
>
  <div
    className={undefined}
  >
    label:
  </div>
  <div
    className={undefined}
  >
    text
  </div>
</div>
`;
